python/lsst/analysis/tools/tasks/objectTableSurveyAnalysis.py

```diff
@@ -0,0 +1,119 @@
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("ObjectTableSurveyAnalysisTask",)
+
+
+from typing import TYPE_CHECKING, Any, Iterable, Mapping, cast
+
+import pandas as pd
+
+if TYPE_CHECKING:
+    from lsst.daf.butler import DataCoordinate, DeferredDatasetHandle
+
+from lsst.pipe.base import connectionTypes as ct
+
+from ..interfaces import KeyedData
+from .base import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
+
+
+class ObjectTableSurveyAnalysisConnections(
+    AnalysisBaseConnections,
+    dimensions=("skymap",),
+    defaultTemplates={"input": "deepCoadd"},
+):
+
+    skymap = ct.Input(
+        doc="Input definition of geometry/bbox and projection/wcs for warped exposures",
+        name="skyMap",
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    data = ct.Input(
+        doc="Input catalog of objects",
+        name="objectTable_tract",
+        storageClass="DataFrame",
+        deferLoad=True,
+        multiple=True,
+        dimensions=("tract", "skymap"),
+    )
+
+
+class ObjectTableSurveyAnalysisConfig(
+    AnalysisBaseConfig, pipelineConnections=ObjectTableSurveyAnalysisConnections
+):
+    pass
+
+
+class ObjectTableSurveyAnalysisTask(AnalysisPipelineTask):
+    ConfigClass = ObjectTableSurveyAnalysisConfig
+    _DefaultName = "objectTableSurveyAnalysis"
+
+    def parsePlotInfo(
+        self, inputs: Mapping[str, Any], dataId: DataCoordinate | None, connectionName: str = "data"
+    ) -> Mapping[str, str]:
+        # Docstring inherited
+
+        if inputs is None:
+            tableName = ""
+            run = ""
+        else:
+            tableName = inputs[connectionName][0].ref.datasetType.name
+            run = inputs[connectionName][0].ref.run
+
+        plotInfo = {"tableName": tableName, "run": run}
+
+        if dataId is not None:
+            for dataInfo in dataId:
+                plotInfo[dataInfo.name] = dataId[dataInfo.name]
+
+        return plotInfo
```

Comment:
Does the plotInfo function in plotUtils not work for this? They look pretty much identical except that the standard one has more information in it.

Reply:
I would have thought that this would come up more regularly, it might be good to generalise the plotUtils one.

Reply:
So, generalize the `parsePlotInfo` in utils, remove the method from `tasks/base.py` and use the utility function both here and in `tasks/base.py`? That sounds like a separate ticket to me.

Reply:
DM-37419

Reply:
I don't think not duplicating code unnesecarily is really another ticket. Having two of these will mean that some plots might end up displaying different things and the plot info generation will get out of sync.

Reply:
There is already code duplication in the `main` branch. The ticket is to deduplicate that. And this `plotInfo` will be a bit different for this task because of multiple tracts in it, and even more so when [DM-37330](https://jira.lsstcorp.org/browse/DM-37330?) lands. It is okay in the object-oriented paradigm for simple and short methods like these to be overridden for special cases, without worrying about partial code duplication. I strongly prefer that over the base method handling all possible cases. Any additions to the base method, that applies to this task as well, has to be added here as well. If we miss that, we will catch it in one of the metrics meetings and can quickly add it here as well.